roles/importer/files/importer/fortiadom5ff/fmgr_getter.py

```diff
@@ -21,23 +21,26 @@ def api_call(url, command, json_payload, sid, show_progress=False, method=''):
     json_payload.update({'method': method})
 
     r = requests.post(url, data=json.dumps(json_payload), headers=request_headers, verify=fwo_globals.verify_certs)
-    if r is None:
+    try:
+        r
+    except Exception as e:
         if 'pass' in json.dumps(json_payload):
-            exception_text = 'error while sending api_call containing credential information to url ' + str(url)
+            exception_text = f'error while sending api_call containing credential information to url {str(url)}'
         else:
-            exception_text = 'error while sending api_call to url ' + str(url) + ' with payload ' + json.dumps(json_payload, indent=2) + ' and  headers: ' + json.dumps(request_headers, indent=2)
-        raise Exception(exception_text)
+            exception_text = f'error while sending api_call to url {str(url)} with payload {json.dumps(json_payload, indent=2)} and  headers: {json.dumps(request_headers, indent=2)}'
+        raise FwApiCallFailed(exception_text)
     result_json = r.json()
     if 'result' not in result_json or len(result_json['result'])<1:
         if 'pass' in json.dumps(json_payload):
-            raise Exception('error while sending api_call containing credential information to url ' + str(url))
+            raise FwApiCallFailed(f'error while sending api_call containing credential information to url {str(url)}')
         else:
             if 'status' in result_json['result'][0]:
-                raise Exception('error while sending api_call to url ' + str(url) + ' with payload ' +
-                        json.dumps(json_payload, indent=2) + ' and  headers: ' + json.dumps(request_headers, indent=2) + ', result=' + json.dumps(r.json()['result'][0]['status'], indent=2))
+                exception_text = f'error while sending api_call to url {str(url)} with payload {json.dumps(json_payload, indent=2)}  and  headers: '
+                exception_text += f'{json.dumps(request_headers, indent=2)}, result={json.dumps(r.json()['result'][0]['status'], indent=2)}'
```

Comment:
```suggestion
                 result_status = r.json()['result'][0]['status']
                exception_text += f'{json.dumps(request_headers, indent=2)}, result={json.dumps(result_status, indent=2)}'
```
Really not sure why, but on my machine I was getting an error `f-string: unmatched '[' (fmgr_getter.py, line 39)` on this line which was fixed with this. Same below